### PR TITLE
[ML] IPP for Registry/Components

### PR DESCRIPTION
**Pull request by needuv**
# Description

This PR adds a new entity definition for intellectual property protection and adds it to registry and component.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/component/e2etests/test_component.py [hunk 1]
```diff
@@ -229,7 +231,7 @@ def test_automl_component(self, client: MLClient, registry_client: MLClient, ran
         )
 
         assert_component_basic_workflow(
-            client=registry_client,
+            client=data_asset_registry_client,
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/component/e2etests/test_component.py:234]: why is this called **data_asset** _registry_client?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/intellectual_property.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml._utils.utils import camel_to_snake
+from azure.ai.ml.constants._common import IPProtectionLevel
+from azure.ai.ml.entities._assets.intellectual_property import IntellectualProperty
+
+
+@experimental
+class IntellectualPropertySchema(metaclass=PatchedSchemaMeta):
+
+    publisher = fields.String()
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/intellectual_property.py:19]: do user have sdk experience like below?
Output(intellectual_property = IntellectualProperty(pulisher=xx)

File: sdk/ml/azure-ai-ml/tests/test_configs/components/basic_component_with_ipp_fields.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+description: This is the basic command component
+tags:
+  tag: tagvalue
+  owner: sdkteam
+
+version: 1
+
+outputs:
+  component_out_path:
+    type: uri_folder
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/test_configs/components/basic_component_with_ipp_fields.yml:15]: add some test case on output has intellectual_property?

Comment (needuv) [reply sdk/ml/azure-ai-ml/tests/test_configs/components/basic_component_with_ipp_fields.yml:15]: yep done

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/intellectual_property.py (hunk 2 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/intellectual_property.py:19]: added

File: sdk/ml/azure-ai-ml/tests/component/e2etests/test_component.py (hunk 1 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/tests/component/e2etests/test_component.py:234]: I'll revert this change

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**